yage-glutin/src/application.rs

```diff
@@ -125,70 +148,136 @@ impl Application {
     }
 
     ///
-    /// Poll events for all windows once.
+    /// Run events loop
+    ///
+    /// Executes the events loop for the application.
+    /// This function will block and run as long as the events loop
+    /// is running. To stop the message loop, stop() has to be called
+    /// on the Application object.
     ///
-    pub fn poll_events(&mut self) {
-        // get references to data we want to access, because closure borrows self
-        let events_loop = &mut self.events_loop;
-        let windows = &self.windows;
+    /// # Returns
+    /// Exit code (0 for no error, > 0 for error)
+    ///
+    pub fn run(&mut self) -> i32 {
+        // Get references to data we want to access, because closure borrows self
+        let windows = &mut self.windows;
         let running = &mut self.running;
-        let first_window = windows.values().next();
+        let proxy = self.events_loop.create_proxy();
+        let mut wakeup_scheduled = false;
 
-        // poll events
-        events_loop.poll_events(|event| {
-            // dispatch event
+        // Run main loop
+        self.events_loop.run_forever(|event| {
+            // [DEBUG]
+            // println!("{:?}", event);
+
+            // Dispatch event
             #[allow(clippy::single_match)]
             match event {
```

Comment:
Something I've been wondering for a while: what's the plan for handling (keyboard, mouse) events on the user side? Might make to add one to the triangle example to try it out 
(I'm on mobile, so I might have overlooked something...)

Reply:
I see no other option than to add an abstraction layer. So the windowing backend (e.g., yage-glutin or yage-web) will take the events, convert them into our own internals events and propagate them to the canvas, and from there on to the renderer and interaction components.